src/air/public_input.cairo

```diff
@@ -0,0 +1,61 @@
+use cairo_verifier::air::public_memory::{
+    Page, PageTrait, ContinuousPageHeader, get_continuous_pages_product
+};
+use cairo_verifier::common::math::{pow, Felt252PartialOrd, Felt252Div};
+
+#[derive(Drop)]
+struct SegmentInfo {
+    // Start address of the memory segment.
+    begin_addr: felt252,
+    // Stop pointer of the segment - not necessarily the end of the segment.
+    stop_ptr: felt252,
+}
+
+#[derive(Drop)]
+struct PublicInput {
```

Comment:
seems like dynamic_params field is missing here right?

Reply:
I will add it either way when i develop public_input_hash

Reply:
Yeah, I only added things that are strictly necessary for AIR and OODS. Feel free to add new fields / methods as you see need while implementing other things